### PR TITLE
Maintain stable root names for heap snapshots

### PR DESCRIPTION
**Pull request by brianrob**
The goal of this change is to ensure that we don't unexpectedly change root names.

 - Detect if the set of roots changes via a test.
 - Add code comments to document these changes via release notes.

### CODE REVIEW

**Review by cincuranet — DISMISSED** [review]

**Comment by leculver** [comment]
This looks right to me, though the reported issue of names changing feels different from what's being fixed here.  They were taking a dependency on some things like "Strong handle" vs "StrongHandle" vs etc, but the code as it exists here wouldn't produce different values.

In short, this change looks like a good addition, definitely check it in...but I feel like we are missing something.

**Review by leculver — APPROVED** [review]

**Comment by brianrob** [comment]
You're right - I did a bit more digging and it looks like the path that creates a gdcump file by listening to ETW specifies a different set of strings.  Let me work to unify this path and then update this PR.

**Comment by brianrob** [comment]
I just added another commit that unifies names for the two paths that create heap snapshots.  This is a breaking change, but should work for tools that are already trying to consume both cases.

File: src/HeapDumpCommon/GCHeapDumpNames.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    internal const string GCHeapsTitle = "GC Heaps";
+    internal const string DotNetRootsTitle = ".NET Roots";
+    
+    internal const string StaticVarsRootTitle = "static vars";
+    internal const string ThreadStaticVarsRootTitle = "thread static vars";
+    internal const string OtherHandlesRootTitle = "other Handles";
+    internal const string DependentHandlesRootTitle = "Dependent Handles";
+    internal const string LocalVarsRootTitle = "local vars";
+    internal const string PinnedLocalVarsRootTitle= "Pinned local vars";
```
Comment (cincuranet) [inline src/HeapDumpCommon/GCHeapDumpNames.cs:11]: nit: Missing space before `=`.

**Review by cincuranet — APPROVED** [review]

**Review by cincuranet — APPROVED** [review]

**Review by asundheim — APPROVED** [review]

**Pull request merged**